benchmarks/persistence/s3_persistence.py

```diff
@@ -0,0 +1,90 @@
+"""Classes for persisting results to a S3-Bucket."""
+
+import json
+import os
+
+import boto3
+import boto3.session
+from attr import define, field
+from attrs.validators import instance_of
+from git import Repo
+
+from benchmarks import Benchmark, Result
+
+
+@define
+class S3ExperimentResultPersistence:
```

Comment:
The more important question is whether we want to hide the method by which an item is created. I see that class more as some kind of an interface implementation (which it was initially) to extend if needed. If we use fsspec and there is an S3 implementation for it, we must define the path explicitly. This necessitates different key constructions, which the function would need to handle. We could build a wrapper around the path construction to access everything as a file (and don't get me wrong, I like that Unix-like idea). However, it seems to me that this might fall under the YAGNI principle. Nevertheless, I will draft somce code, as there might be something I am overlooking.

Reply:
@AdrianSosic I've added some changes as discussed. Let's have another refinement :)

Reply:
Is this comment still relevant @AdrianSosic @fabianliebig ? There is another thread regarding the usage of `fsspec` after this refactoring which replaced inheritane by protocols if I understand it correctly - so is there anything related to just this comment here or can it be resolved?

Reply:
I think that can be closed and we will discuss further thinks down below.